### PR TITLE
Notifications mgmt render fixes

### PR DESCRIPTION
**Pull request by mhagel**
## Link to Issue
Closes: #3015 

## Description of Changes
Resolves from issue description:
- [x] Multiline comments don't render well,
- [x] removing a subscription does not update the page
- [x] receive emails checkbox does not work
- [x] In-app toggle does not work
- [x] clicking item is taking user to dead end page (probably missing scope prefix in navigate function)

## Test Plan
- CA (click around) tested on local and frack:
  - http://localhost:8080/notification-settings
  - click notification title to confirm correct navigation
  - http://localhost:8080/dydx/new/proposal renders with log in/out (made a tweak here)

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

File: packages/commonwealth/client/scripts/identifiers.ts [hunk 1]
```diff
@@ -24,10 +24,11 @@ export const getProposalUrlPath = (
   } else {
     basePath = `/proposal/${id}`;
   }
-  if (omitActiveId || (app.isCustomDomain() && !chainId)) {
```
Comment (mhagel) [inline packages/commonwealth/client/scripts/identifiers.ts:27]: this logic prevented the chainID condition form hitting. Please LMK if my revision doesn't make sense. This function handles a lot of cases I think.

File: packages/commonwealth/client/styles/pages/notification_settings/helper_components.scss [hunk 2]
```diff
@@ -45,6 +45,7 @@
 
     .subscription-body-text.Text {
       color: $neutral-600;
+      font-size: 14px;
```
Comment (mhagel) [inline packages/commonwealth/client/styles/pages/notification_settings/helper_components.scss:48]: matches Prod, but LMK if 12px is preferred

File: packages/commonwealth/client/scripts/views/pages/new_proposal/index.tsx [hunk 3]
```diff
@@ -19,34 +19,29 @@ import { AaveProposalForm } from './aave_proposal_form';
...
-
       setIsLoaded(app.chain.loaded);
     });
   }, [app.chain?.loaded]);
 
   useEffect(() => {
-    app.loginStateEmitter.on('redraw', () => {
-      console.log('app.isLoggedIn()', app.isLoggedIn());
-      setIsLoggedIn(app.isLoggedIn());
-    });
```
Comment (mhagel) [inline packages/commonwealth/client/scripts/views/pages/new_proposal/index.tsx:46]: Now that I know `forceRerender` is a thing, I am using it on this change I made few days ago. I tested to make sure behavior is the same.

File: packages/commonwealth/client/scripts/views/pages/notification_settings/helper_components.tsx [hunk 4]
```diff
@@ -51,9 +53,7 @@ const getTextRows = (
           </CWText>
         </div>
         <CWText type="caption" className="subscription-body-text" noWrap>
-          {renderQuillTextBody(subscription.Thread.body, {
-            hideFormatting: true,
-          })}
```
Comment (mhagel) [inline packages/commonwealth/client/scripts/views/pages/notification_settings/helper_components.tsx:56]: This caused some wacky rendering. From Prod, it looks like we only need the first line, and no adherence to formatting. LMK if i'm wrong about that.

File: packages/commonwealth/client/scripts/identifiers.ts (hunk 1 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/identifiers.ts:27]: I would not modify this function as it is used in many places. in `getTextRows` there is a `setRoute(threadUrl)` which works like navigate from useCommonNavigate. Check if passing third argument to setRoute as chain id would do the work?

File: packages/commonwealth/client/scripts/views/pages/notification_settings/helper_components.tsx [hunk 5]
```diff
@@ -31,6 +31,8 @@ const getTextRows = (
       subscription.Chain.id
     );
 
+    const firstLine = JSON.parse(subscription.Thread.body)?.ops[0]?.insert;
```
[the remark this answers is not in the document]
Comment (mhagel) [reply packages/commonwealth/client/scripts/views/pages/notification_settings/helper_components.tsx:34]: `plaintext` is much better, thank you

File: packages/commonwealth/client/scripts/identifiers.ts (hunk 1 above)
Comment (mhagel) [reply packages/commonwealth/client/scripts/identifiers.ts]: Agree, it is not worth risking damage to other use cases. I made a new function for this case: `getNotificationUrlPath`

File: packages/commonwealth/client/scripts/views/pages/view_thread/index.tsx [hunk 6]
```diff
@@ -329,7 +329,7 @@ const ViewThreadPage = ({ identifier }: ViewThreadPageProps) => {
       });
 
       app.profiles.isFetched.on('redraw', () => {
-        if (!prefetch[threadIdAndType]['profilesFinished']) {
+        if (!prefetch[threadIdAndType]?.['profilesFinished']) {
```
Comment (mhagel) [inline packages/commonwealth/client/scripts/views/pages/view_thread/index.tsx:332]: null checks to resolve a crash that sometimes happens

**Review by masvelio — COMMENTED** [review]
1. toggle did not work for me
2. Eg in ethereum I could not click the item. In Affiliate DAO it was clickable pretty much every time but in dydx I had problems with clicking as well. 

https://user-images.githubusercontent.com/14819225/225898129-31f11f2b-1ada-4f6c-9d2c-fc7bb39c47d4.mov

**Comment by mhagel** [comment]
> 1. toggle did not work for me
> 2. Eg in ethereum I could not click the item. In Affiliate DAO it was clickable pretty much every time but in dydx I had problems with clicking as well.

@masvelio 
1. toggle animation is resolved by latest from `react`
2. Good catches. I have resolved the "New Threads" case, but I am having trouble reproducing the last example in the video.

**Review by masvelio — APPROVED** [review]
💪

**Pull request merged**